CHALLENGE.md

```diff
@@ -52,71 +51,71 @@ The project includes:
+- When "OK" is selected, the tournament is deleted immediately in the UI using an optimistic update without any loading indicators and a fetch call to the fake REST API is made.
 
 ### 4. Search tournaments
 
 Requirements:
 
-- The search should not be performed against the local data but by calling the endpoint with the search term.
-- The search should be performed on the user's typing and not by hitting the Enter key.
+- The search function should call the endpoint with the search term, rather than searching through local data.
+- The search should trigger on user input, not by pressing the Enter key.
 
-#### 4.A) Add a `Search tournament ...` input field:
+#### 4.A) Add a search input field labeled `Search tournament ...`:
 
 ![search-input](.github/search-input.png)
 
-#### 4.B) Show `Loading tournaments ...` while fetching:
+#### 4.B) Display `Loading tournaments ...` while data is being fetched:
 
 ![search-loading](.github/search-loading-state.png)
 
-#### 4.C) Show `Something went wrong.` with a `RETRY` button when the fetching has failed:
+#### 4.C) Display `Something went wrong.` with a `RETRY` if the data fetch fails:
 
 ![search-error](.github/search-error-state.png)
 
-- Clicking on the `RETRY` button will retry the fetching.
+- Pressing the `RETRY` button retries the data fetch.
 
-#### 4.D) Show all tournaments from the search result when the fetching has succeed:
+#### 4.D) Display all tournaments from the search results if the data fetch is successful:
 
 ![search-success](.github/search-success-state.png)
 
-#### 4.E) Show `No tournaments found.` when the search result is empty:
+#### 4.E) Show `No tournaments found.` if the search result is empty:
 
 ![search-no-result](.github/search-no-result-state.png)
 
-#### 4.F) Search functionality should be optimized with performance in mind.
+#### 4.F) Optimize the search functionality for performance.
 
 ### 5. Create a tournament
 
 ![create-tournament](.github/create-tournament.png)
 
-- Add a `CREATE TOURNAMENT` button.
+- Add a button labeled `CREATE TOURNAMENT`.
 
 ![create-tournament-prompt](.github/create-tournament-prompt.png)
 
-- Clicking on the `CREATE TOURNAMENT` button will open a browser prompt with a message `Tournament Name:`, an input field to enter the tournament name and the buttons `Cancel` to cancel and `OK` to confirm.
+- Pressing the `CREATE TOURNAMENT` button opens a browser prompt with the message `Tournament Name:`, an input field to enter the tournament name, and the buttons `Cancel` to cancel and `OK` to confirm.
 - The tournament name must contain only Latin letters, numbers, and spaces, not an empty string or only spaces.
-- When confirming, the tournament will be created on the fake REST API and added on the UI without any loading indicators.
+- When `OK` is selected, the tournament is created on the simulated REST API and added to the top of the UI without any loading indicators.
```

Comment:
`to the top of the UI` sounds a bit strange..
Maybe `..at the start of the tournament list`, something like that